tree/tree/src/TTree.cxx

```diff
@@ -1209,6 +1209,19 @@ void TTree::AddClone(TTree* clone)
    }
 }
 
+void ThrowIfReshuffled(TTree *t1, TTree *t2)
+{
+   const auto isT1Reshuffled = t1->TestBit(TTree::kEntriesReshuffled);
+   const auto isT2Reshuffled = t2->TestBit(TTree::kEntriesReshuffled);
+   if (isT1Reshuffled || isT2Reshuffled) {
+      const std::string reshuffledTreeName = isT1Reshuffled ? t1->GetName() : t2->GetName();
+      const std::string msg = "Tree " + reshuffledTreeName +
+                              " has kEntriesReshuffled set, and cannot be used as friend nor can be added as a "
+                              "friend. Unset the bit manually if you know what you are doing.";
+      throw std::invalid_argument(msg);
```

Comment:
Throwing here is (at least in the current state of the code) invalid.   It leaves the TTree is a weird/broken state as the calling function is not exception safe.

Reply:
You are completely right, I did not think about that. I think we could switch to `Error` or `Warning` then -- what do you think?